src/components/AssetSearch/AssetSearch.tsx

```diff
@@ -70,10 +70,9 @@ export class AssetSearch extends React.Component<
       metadata:
         (query.advancedSearch &&
           query.advancedSearch.metadata &&
-          query.advancedSearch.metadata.reduce(
-            (a, c) => ({ ...a, [c.key]: c.value }),
-            {}
-          )) ||
+          query.advancedSearch.metadata
+            .filter(meta => meta.key !== '')
+            .reduce((a, c) => ({ ...a, [c.key]: c.value }), {})) ||
```

Comment:
I would change it a bit. If you use `filter` and `reduce` you'll get `O(2n)` complexity. But you can avoid filtering by moving that logic to reduce callback (to acheave `O(n)` complexity):

```typescript
.reduce((a, c) => (c.key ? { ...a, [c.key]: c.value } : a), {}))
```
It's not so important in this case, but its better to think about complexity when you're formatting/modifining data

Reply:
Complexity reductioned on both master and sdk-2.0 branches PRs.